drivers/shared/src/make_dmy.c

```diff
@@ -66,193 +61,49 @@ make_dmy(global_param_struct *global)
 
-    /** Create Date Structure for each Modeled Time Step **/
-    jday = day;
-    if (LEAPYR(year)) {
-        days[1] = 29;
-    }
-    else {
-        days[1] = 28;
-    }
-    for (ii = 0; ii < month - 1; ii++) {
-        jday += days[ii];
-    }
-
-    DONE = false;
-    ii = 0;
-
-    while (!DONE) {
-        temp[ii].hour = hr;
-        temp[ii].day = day;
-        temp[ii].month = month;
-        temp[ii].year = year;
-        temp[ii].day_in_year = jday;
-
-        get_next_time_step(&year, &month, &day, &hr, &jday, global->dt);
-
-        ii++;
-        if (ii == global->nrecs) {
-            DONE = true;
-        }
-    }
-
-    /** Determine number of forcing records to skip before model start time **/
-    for (i = 0; i < 2; i++) {
-        if (param_set.FORCE_DT[i] != MISSING) {
-            if (global->forceyear[i] > 0) {
-                tmpyear = global->forceyear[i];
-                tmpmonth = global->forcemonth[i];
-                tmpday = global->forceday[i];
-                tmphr = global->forcehour[i];
-                tmpjday = tmpday;
-                if (LEAPYR(tmpyear)) {
-                    days[1] = 29;
-                }
-                else {
-                    days[1] = 28;
-                }
-                for (ii = 0; ii < tmpmonth - 1; ii++) {
-                    tmpjday += days[ii];
-                }
-
-                while (tmpyear < temp[0].year ||
-                       (tmpyear == temp[0].year && tmpjday <
-                        temp[0].day_in_year) ||
-                       (tmpyear == temp[0].year && tmpjday ==
-                        temp[0].day_in_year && tmphr < temp[0].hour)) {
-                    get_next_time_step(&tmpyear, &tmpmonth, &tmpday, &tmphr,
-                                       &tmpjday, global->dt);
-
-                    global->forceskip[i]++;
-                }
-                global->forceoffset[i] = global->forceskip[i];
-            }
-        }
+    /** Create Date Structure for each Modele Time Step **/
+    for (i = 0, numdate = start_num;
+         i < global->nrecs;
+         i++, numdate += dt_days) {
+        num2date(global->time_origin_num, numdate, 0., global->calendar,
+                 TIME_UNITS_DAYS, &temp[i]);
     }
 
-    /** Determine the number of records to skip before starting output files **/
-    skiprec = 0;
-    for (i = 0; i < global->skipyear; i++) {
-        if (LEAPYR(temp[skiprec].year)) {
-            skiprec += DAYS_PER_LYEAR * HOURS_PER_DAY / global->dt;
-        }
-        else {
-            skiprec += DAYS_PER_YEAR * HOURS_PER_DAY / global->dt;
-        }
-    }
-    global->skipyear = skiprec;
-
     return temp;
 }
 
 /******************************************************************************
- * @brief    Get the next timestep.
- *****************************************************************************/
-void
-get_next_time_step(unsigned short *year,
```

Comment:
Replaced by something else.  I have a calendar aware implementation that uses `num2date()`.